js/layout.js

```diff
@@ -108,13 +113,40 @@ function updatePageLayout(updateType) {
 }
 
 function refitTerminal() {
+    // Custom function to replace the terminal refit function as it was a bit buggy
+
     // Re-fitting the terminal requires a full re-layout of the DOM which can be tricky to time right.
     // see https://www.macarthur.me/posts/when-dom-updates-appear-to-be-asynchronous
     window.requestAnimationFrame(() => {
         window.requestAnimationFrame(() => {
             window.requestAnimationFrame(() => {
-                if (state.fitter) {
-                    state.fitter.fit();
+                const TERMINAL_ROW_HEIGHT = state.terminal._core._renderService.dimensions.css.cell.height;
+                const TERMINAL_COL_WIDTH = state.terminal._core._renderService.dimensions.css.cell.width;
+
+                // Get the height of the header, footer, and serial bar to determine the height of the terminal
+                let siteHeader = document.getElementById('site-header');
+                let mobileHeader = document.getElementById('mobile-header');
+                let headerHeight = siteHeader.offsetHeight;
+                if (siteHeader.style.display === 'none') {
+                    headerHeight = mobileHeader.offsetHeight;
+                }
+                let foorterBarHeight = document.getElementById('footer-bar').offsetHeight;
```

Comment:
```suggestion
                let footerBarHeight = document.getElementById('footer-bar').offsetHeight;
```

Reply:
Good catch. I also need to update line 136.